lib/rspec/json_expectations/matchers.rb

```diff
@@ -55,6 +55,64 @@ def traverse(expected, actual, negate=false)
   end
 end
 
+RSpec::Matchers.define :include_unordered_json do |expected|
+
+  # RSpec 2 vs 3
+  if respond_to?(:failure_message)
+    match do |actual|
+      traverse(expected, actual, false)
+    end
+
+    match_when_negated do |actual|
+      traverse(expected, actual, true)
+    end
+
+    failure_message do |actual|
+       RSpec::JsonExpectations::FailurePresenter.render(@include_json_errors)
+    end
+
+    failure_message_when_negated do |actual|
+       RSpec::JsonExpectations::FailurePresenter.render(@include_json_errors)
+    end
+  else
+    match_for_should do |actual|
+      traverse(expected, actual, false)
+    end
+
+    match_for_should_not do |actual|
+      traverse(expected, actual, true)
+    end
+
+    failure_message_for_should do |actual|
+      RSpec::JsonExpectations::FailurePresenter.render(@include_json_errors)
+    end
+
+    failure_message_for_should_not do |actual|
+      RSpec::JsonExpectations::FailurePresenter.render(@include_json_errors)
+    end
+  end
+
+  def traverse(expected, actual, negate=false)
```

Comment:
Seems, like this function can become a TemplateMethod on some module, that can be called with a few arguments from here and from other matcher.